src/main/java/com/uid2/client/PublisherUid2Helper.java

```diff
@@ -0,0 +1,129 @@
+package com.uid2.client;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonObject;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.security.SecureRandom;
+import java.time.Instant;
+import java.util.Arrays;
+
+
+public class PublisherUid2Helper {
+    /**
+     * @param base64SecretKey your UID2 client secret
+     */
+    public PublisherUid2Helper(String base64SecretKey) {
+        secretKey = InputUtil.base64ToByteArray(base64SecretKey);
+    }
+
+    /**
+     * @param tokenGenerateInput represents the input required for <a href="https://github.com/UnifiedID2/uid2docs/blob/main/api/v2/endpoints/post-token-generate.md#unencrypted-json-body-parameters">/token/generate</a>
+     * @return an EnvelopeV2 instance to use in the POST body of <a href="https://github.com/UnifiedID2/uid2docs/blob/main/api/v2/endpoints/post-token-generate.md">/token/generate</a>
+     */
+    public EnvelopeV2 createEnvelopeForTokenGenerateRequest(TokenGenerateInput tokenGenerateInput) {
+        final int nonceLength = 8;
+        byte[] nonce = new byte[nonceLength];
+        secureRandom.nextBytes(nonce);
+        return createEnvelopeImpl(tokenGenerateInput, nonce, Instant.now(), null);
+    }
+
+    /**
+     * @param response the response body returned by a call to <a href="https://github.com/UnifiedID2/uid2docs/blob/main/api/v2/endpoints/post-token-generate.md">/token/generate</a>
+     * @param envelope the EnvelopeV2 instance returned by {@link #createEnvelopeForTokenGenerateRequest}
+     * @return an IdentityTokens instance
+     */
+    public IdentityTokens createIdentityfromTokenGenerateResponse(String response, EnvelopeV2 envelope) {
+        String identityJsonString = decrypt(response, secretKey, false, envelope.getNonce());
+        JsonObject jsonIdentity = new Gson().fromJson(identityJsonString, JsonObject.class);
```

Comment:
Probably could be called responseJson or something? The identity part is in the body.

Reply:
sure, makes sense